tests/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+use std::path::PathBuf;
+use anyhow::{bail, Result};
+use map_model::Map;
+use abstio::MapName;
+use abstutil::Timer;
+use prettydiff::text::diff_lines;
+
+/// Run the contents of a .osm through the full map importer with default options.
+pub fn import_map(path: String) -> Map {
+    let mut timer = Timer::new("convert synthetic map");
+    let name = MapName::new("zz", "oneshot", &abstutil::basename(&path));
+    let clip = None;
+    let raw = convert_osm::convert(
+        path,
+        name,
+        clip,
+        convert_osm::Options::default(),
+        &mut timer,
+    );
+    Map::create_from_raw(raw, map_model::RawToMapOptions::default(), &mut timer)
+}
+
+/// Obtains a path to a test file (test code only!)
+/// This is a convenience function for writing test code. It allow tests code from anywhere in the
+/// workspace to access test files (eg input .osm files, golden outputfiles etc) which are stored within
+/// the `tests` package.
+/// This function make direct reference to the location of this source file (using the `file!()` marco)
+/// and hence should only be used in test code and not in any production code.
+pub fn get_test_file_path(path: String) -> Result<String, String> {
```

Comment:
There may be a simpler way to solve the problem of finding test resources... not sure though. The only thing that comes to mind is the `include_str!` macro. The total is 150KB right now; I think I've abused the include macros up to triple-digit megabytes before ;)  The resulting executable size is larger, but for test-only things, that doesn't matter much. It does mean you have to recompile when you change the test data, though. Up to you on this -- if you're happy with this approach, fine by me

Reply:
Oh wait, you do need a path, because you're feeding this into the map importer process. Ignore.